src/main/java/com/appoptics/integrations/kafka/broker/KafkaAppopticsReporter.java

```diff
@@ -17,18 +20,46 @@
     private Reporter.Builder reporterBuilder;
     private Reporter reporter;
 
+    private final String URL = "appoptics.url";
+    private final String TOKEN = "appoptics.token";
+    private final String AGENT_IDENTIFIER = "appoptics.agent.identifier";
+    private final String TAGS = "appoptics.tags";
+
     @Override
     public synchronized void init(VerifiableProperties props) {
-        String apiUrl = props.getString("appoptics.url");
+        String apiUrl = props.getString(URL);
         if (apiUrl == null) {
             apiUrl = "https://api.appoptics.com/v1/measurements";
         }
 
-        String token = props.getString("appoptics.token");
-        String source = props.getString("appoptics.agent.identifier");
+        String token = props.getString(TOKEN);
+        List<Tag> tags = new ArrayList<Tag>();
+
+        String source = props.getString(AGENT_IDENTIFIER, "");
+        if (!source.isEmpty()) {
+            tags.add(new Tag("source", source));
+        }
+
+        String customTags = props.getString(TAGS, "");
+        if (!customTags.isEmpty()) {
+            String[] str = new String[]{customTags};
+            if (customTags.contains(",")) {
+                str = customTags.split(",");
+            }
+            for (String tagString : str) {
+                if (customTags.contains(":")) {
+                    String properties[] = tagString.split(":");
+                    if (str.length == 2) {
```

Comment:
`:` is valid inside a tag name or value, should we use a separator here that isn't valid?

Reply:
@petercable same as below, I need additional information thats currently not available here https://docs.appoptics.com/api/#create-a-measurement